.github/workflows/test.yaml

```diff
@@ -0,0 +1,74 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - '**/*'
+
+jobs:
+  test-android-sdk:
+    runs-on: macos-latest
+    env:
+      ANDROID_API_LEVEL: 33
+    steps:
+      - name: Check out Java SDK
+        uses: actions/checkout@v3
+        with:
+          repository: 'Eppo-exp/android-sdk'
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+
+      - name: 'Set up GCP SDK'
+        uses: 'google-github-actions/setup-gcloud@v0'
+
+      - name: Restore gradle.properties
+        env:
+          MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
+        shell: bash
+        run: |
+          mkdir -p ~/.gradle/
+          echo "GRADLE_USER_HOME=${HOME}/.gradle" >> $GITHUB_ENV
+          echo "MAVEN_USERNAME=${MAVEN_USERNAME}" > ~/.gradle/gradle.properties
+          echo "MAVEN_PASSWORD=${MAVEN_PASSWORD}" >> ~/.gradle/gradle.properties
+
+      - name: Set up test data
+        run: make test-data
+
+      - name: Spin up emulator and run tests
+        id: testing
+        uses: ReactiveCircus/android-emulator-runner@v2
+        with:
+          api-level: 33
+          target: google_apis
+          arch: x86_64
+          emulator-options: -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim -dns-server 8.8.8.8
+          script: |
+            echo "Emulator started" 
+            adb logcat -c                                     # clear logs
+            mkdir -p app/                                     # create directory
+            touch app/emulator.log                            # create log file
+            chmod 777 app/emulator.log                        # allow writing to log file
+            adb logcat | grep EppoSDK >> app/emulator.log &   # pipe all logcat messages into log file as a background process
```

Comment:
We use `grep` so we get all the logs from our SDK, regardless of the specific class logging.